test/steps/v2_steps.py

```diff
@@ -2264,3 +2340,322 @@ class TestCase(DryrunTestCaseMixin, unittest.TestCase):
 
     ts.assertNoError(drr)
     ts.assertLocalStateContains(drr, account, dict(key=key, value=val))
+
+
+@given("a new AtomicTransactionComposer")
+def create_atomic_transaction_composer(context):
+    context.atomic_transaction_composer = (
+        atomic_transaction_composer.AtomicTransactionComposer()
+    )
+
+
+@when("I make a transaction signer for the {account_type} account.")
+def create_transaction_signer(context, account_type):
+    if account_type == "transient":
+        private_key = context.transient_sk
+    elif account_type == "signing":
+        private_key = mnemonic.to_private_key(context.signing_mnemonic)
+    else:
+        raise NotImplementedError(
+            "cannot make transaction signer for " + account_type
+        )
+    context.transaction_signer = (
+        atomic_transaction_composer.AccountTransactionSigner(private_key)
+    )
+
+
+@when('I create the Method object from method signature "{method_signature}"')
+def build_abi_method(context, method_signature):
+    context.abi_method = abi.Method.from_signature(method_signature)
+
+
+@when("I create a transaction with signer with the current transaction.")
+def create_transaction_with_signer(context):
+    context.transaction_with_signer = (
+        atomic_transaction_composer.TransactionWithSigner(
+            context.transaction, context.transaction_signer
+        )
+    )
+
+
+@when("I add the current transaction with signer to the composer.")
+def add_transaction_to_composer(context):
+    context.atomic_transaction_composer.add_transaction(
+        context.transaction_with_signer
+    )
+
+
+def split_and_process_abi_args(method, arg_string):
+    method_args = []
+    arg_tokens = arg_string.split(",")
+    arg_index = 0
+    for arg in method.args:
+        # Skip arg if it does not have a type
+        if isinstance(arg.type, abi.ABIType):
+            arg = arg.type.decode(base64.b64decode(arg_tokens[arg_index]))
+            method_args.append(arg)
+            arg_index += 1
+    return method_args
+
+
+@when("I create a new method arguments array.")
+def create_abi_method_args(context):
+    context.method_args = []
+
+
+@when(
+    "I append the current transaction with signer to the method arguments array."
+)
+def append_txn_to_method_args(context):
+    context.method_args.append(context.transaction_with_signer)
+
+
+@when(
+    'I append the encoded arguments "{method_args:MaybeString}" to the method arguments array.'
+)
+def append_app_args_to_method_args(context, method_args):
+    # Returns a list of ABI method arguments
+    app_args = split_and_process_abi_args(context.abi_method, method_args)
```

Comment:
This works for now so it doesn't need to be changed in this PR, but I'd like this step to not assume that `method_args` match the arguments from `context.abi_method`.

Meaning it should be possible to call this step many times to append arguments, and then only in the `I add a method call...` step should the arguments be decoded. This will give us more flexibility to mix transaction args and normal args, like `(uint64,pay,uint64,axfer,uint64)`.

This is how I did in the JS SDK: https://github.com/algorand/js-algorand-sdk/blob/6cb22886b089544349448bc1097164dc70a085b8/tests/cucumber/steps/steps.js#L4750-L4791

Reply:
That's a good point, we should make the step as general as possible.